homeassistant/components/wilight/switch.py

```diff
@@ -112,46 +90,48 @@ async def async_setup_entry(
 
     # Handle services for a discovered WiLight device.
     async def set_watering_time(entity, service: Any) -> None:
-        entity_id = service.data[ATTR_ENTITY_ID]
-
-        if (entity.entity_id == entity_id) & isinstance(entity, WiLightValveSwitch):
-            watering_time = service.data[ATTR_WATERING_TIME]
-            await entity.async_set_watering_time(watering_time=watering_time)
+        if not isinstance(entity, WiLightValveSwitch):
+            raise ValueError("Entity is not an instance of WiLightValveSwitch")
```

Comment:
These messages are user facing so we should use a language that users can understand. The message should explain what is incorrect in the user input, in this case the entity targeted.
```suggestion
            raise ValueError("Entity is not a WiLight valve switch")
```